src/core/aws-utils.ads

```diff
@@ -80,20 +80,33 @@ package AWS.Utils is
    --  General helper functions --
    -------------------------------
 
-   function Random return Random_Integer;
-   --  Returns a random integer number
-
-   procedure Random_String (Item : out String) with
-     Post => (for all C of Item => C in '0' .. '9' | 'a' .. 'z' | 'A' .. 'Z');
+   function Random (Task_Safe : Boolean := False) return Random_Integer;
+   --  Returns a random integer number.
+   --  Task_Safe parameter need to avoid concurrency on modifying random
+   --  generator internal state. Note that Task_Safe = False does not mean
```

Comment:
I see, so my comment about have safe by default is maybe not an issue.

Reply:
I read this after I moved Task_Safe default to True. Maybe we need global flag for this mode. If anybly want to detect races, he can set it to True.

Reply:
But it is not about "detecting races" just to satisfy a tool. It is about avoiding them in the first place. AWS doesn't have optional tasks, so the code needs to decide for every call whether there are multiple tasks involved or not. A global default doesn't really make sense here.

Reply:
I created separate routine to Set_Random_Task_Safe and removed Task_Safe parameter from random routines. Entire commit bacame smaller.

Reply:
<<It is about avoiding them in the first place.>> GNAT random generator is OK with concurrently update of  the internal state. If you need to satisfy the race detection tools you can use Set_Random_Task_Safe routine.

Reply:
I don't understand how you come to the conclusion that the random generator is task safe. AWS uses Ada.Numerics.Discrete_Random, for which the generator is from System.Random_Numbers. There are no protected types, nor atomic types, nor mutexes in that package, so I do not see how it would be safe to use from multiple tasks. Can you explain ?

Reply:
it is not really task safe. Internal state mixed when modified concurrently, but it is not lead to failure in random numbers generation. The random numbers became a little bit more random.

Reply:
Actually no, they just become garbage. For instance, they no longer follow the ARM requirement that they be uniformly distributed. For all I know, we might end up with invalid values in the generator which means that from now on we are getting the same number always.
Anything that is not task safe must be protected in the context of AWS, which uses tasking heavily and on multiple platforms. Thread-sanitizer is an excellent tool to find such cases where things seem to work but only work by chance and could break when moving to new versions of the system, new platform, new compiler, or even new phase of the moon.

Reply:
You are right, I just made experiment and I see that the concurrency lead to the repeating of the same numbers is sequence.
I'll chenge the implementation to always protected.

Reply:
Thanks Dmitriy !